devops/containers/ubuntu2404_build.Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM nvidia/cuda:12.6.3-devel-ubuntu24.04
```

Comment:
The Ubuntu 22 docker container uses CUDA 12.1.0 (https://github.com/intel/llvm/blob/sycl/devops/containers/ubuntu2204_build.Dockerfile#L1). Are we good to use CUDA 12.6.3 with DPC++?
Similarly, Ubuntu 22 container uses ROCM 6.1.1 while Ubuntu 24 uses ROCM 6.3.

Reply:
Good question. Based on the Codeplay [doc](https://developer.codeplay.com/products/oneapi/nvidia/2025.0.0/guides/get-started-guide-nvidia) (who owns the AMD/NVidia support), they say CUDA12+ works but isn't they version they are currently testing on:

![image](https://github.com/user-attachments/assets/e411eef0-0d3d-43a4-ab91-553c0e76ec27)

And for ROCM:

![image](https://github.com/user-attachments/assets/baf4f2cb-bfd5-479c-bad3-f18d8bdccac1)

Here it doesn't list 6.3 specifically but says a wide variety should work. Also one of the runners is already using an unsupported GPU so it's not like we are following this doc today.

So IMO based on the doc I think it's fine to try to use the newer API versions but we can revert if there's some problem.

Reply:
Seems reasonable to me. @npmiller @JackAKirk FYI.

Reply:
I'm going to merge this for now. This PR doesn't make us use the new versions in testing yet, it just adds the Docker images. There will be a subsequent PR to move the CI over to use those new images. Happy to address any feedback from the Codeplay team.